pkg/kademlia/workers_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package kademlia
+
+import (
+	"context"
+	"net"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"storj.io/storj/internal/mock"
+	"storj.io/storj/pkg/dht/mocks"
+	"storj.io/storj/pkg/node"
+	"storj.io/storj/pkg/provider"
+	proto "storj.io/storj/protos/overlay"
+)
+
+var (
+	ctx = context.Background()
+)
+
+func TestGetWork(t *testing.T) {
+	cases := []struct {
+		name     string
+		worker   *worker
+		expected *proto.Node
+		ch       chan *proto.Node
+	}{
+		{
+			name:     "test valid chore returned",
+			worker:   newWorker(context.Background(), nil, []*proto.Node{&proto.Node{Id: "1001"}}, nil, node.StringToID("1000"), 5),
+			expected: &proto.Node{Id: "1001"},
+			ch:       make(chan *proto.Node, 2),
+		},
+		{
+			name: "test no chore left",
+			worker: func() *worker {
+				w := newWorker(context.Background(), nil, []*proto.Node{&proto.Node{Id: "foo"}}, nil, node.StringToID("foo"), 5)
+				w.maxResponse = 0
+				w.pq.Pop()
+				assert.Len(t, w.pq, 0)
+				return w
+			}(),
+			expected: nil,
+			ch:       make(chan *proto.Node, 2),
+		},
+	}
+
+	for _, v := range cases {
+		ctx, cf := context.WithTimeout(context.Background(), 50*time.Millisecond)
+		defer cf()
+
+		v.worker.cancel = cf
+		v.worker.getWork(ctx, v.ch)
+
+		if v.expected != nil {
+			actual := <-v.ch
+			assert.Equal(t, v.expected, actual)
+		} else {
+			assert.Len(t, v.ch, 0)
+		}
+
+	}
+}
+
+func TestWorkerLookup(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockDHT := mock_dht.NewMockDHT(ctrl)
+	mockRT := mock_dht.NewMockRoutingTable(ctrl)
+
+	lis, err := net.Listen("tcp", ":0")
```

Comment:
`127.0.0.1:0`

Reply:
why?

Reply:
I agree with @coyle, we should test for both IPv4 and IPv6

Reply:
Because `:0` triggers firewalls on Windows, which makes for a bad development experience. I agree that IPv6 should be tested, but does not test it.

Reply:
I think the best solution is to not use windows ;)

Reply:
Good to know though. I'll change these

Reply:
Can we include the IPv6 Test with [::1]:0 as addition to 127.0.0.1:0?

Reply:
Ideally we would have integration tests, that test both separately and one together. I created an issue for it https://storjlabs.atlassian.net/browse/V3-436